src/web/TextInput.tsx

```diff
@@ -295,6 +321,62 @@ export class TextInput extends React.Component<Types.TextInputProps, TextInputSt
         }
     }
 
+    private static _shouldAutoResize(props: Types.TextInputProps) {
+        // Single line boxes don't need auto-resize
+        if (!props.multiline) {
+            return false;
+        }
+
+        const combinedStyles = Styles.combine(props.style);
+        if (!combinedStyles || typeof combinedStyles === 'number') {
+            // Number-type styles aren't allowed on web but if they're found we can't decode them so assume not fixed height
+            return true;
+        } else if (Array.isArray(combinedStyles)) {
+            // Iterate across the array and see if there's any height value
+            return combinedStyles.some(style => {
```

Comment:
This check is a bit fragile. There are multiple ways to fixing the height of a view other than setting a "height" attribute. For example, you can specify absolute positioning with a top and bottom value. This is probably sufficient in practice, but it deserves a comment at least.